// dummy-module/additional.js
"use strict";

var
    p3 = 0,
    array = [],
    //Dummy = function () {
    //    if (Dummy.prototype._singletonInstance) {
    //        return Dummy.prototype._singletonInstance;
    //    }
    //
    //    Dummy.prototype._singletonInstance = this;
    //
    //    this.p1 = undefined;
    //    var p2 = undefined;
    //
    //    this.setP1 = function (value) {
    //        this.p1 = value;
    //    };
    //
    //    this.getP1 = function () {
    //        return this.p1;
    //    };
    //
    //    this.setP2 = function (value) {
    //        p2 = value;
    //    };
    //
    //    this.getP2 = function () {
    //        return p2;
    //    };
    //
    //    this.setP3 = function (value) {
    //        p3 = value;
    //    };
    //
    //    this.getP3 = function () {
    //        return p3;
    //    };
    //
    //},
    cachedDummy = {

        setP1: function(value) {

        },
        getP1: function(value) {

        },

        setP3: function (value) {
            p3 = value;
        },
        getP3: function() {
            return p3;
        }
    },
    f = function(){
        p3++;
    },
    getP3 = function(){
        return p3;
    };



module.exports = {
    f: f,
    getP3: getP3,
    array: array
};
